unlock-js/src/deploy.js

```diff
@@ -1,33 +1,31 @@
-const Web3 = require('web3')
+const ethers = require('ethers')
 const gas = require('./constants').GAS_AMOUNTS
 
 export default async function deploy(
   host,
   port,
   Unlock,
-  onNewContractInstance = () => {},
-  web3Object
+  onNewContractInstance = () => {}
 ) {
-  const web3 = web3Object || new Web3(`http://${host}:${port}`)
-  const unlock = new web3.eth.Contract(Unlock.abi)
+  const provider = new ethers.providers.JsonRpcProvider(
+    `http://${host}:${port}`
+  )
+  // Load the wallet to deploy the contract with
+  const wallet = provider.getSigner(0)
+  const factory = new ethers.ContractFactory(
+    Unlock.abi,
+    Unlock.bytecode,
+    wallet
+  )
+  const accounts = await provider.listAccounts()
+  const unlockContract = await factory.deploy({ gasLimit: gas.deployContract })
+  onNewContractInstance(unlockContract)
```

Comment:
Shouldn't this be after `await unlockContract.deployed()` to make sure that the callback is only called after the contract has effectively been deplpyed?

Reply:
or even after `initialize`?

Reply:
That is fine if you'd like. The current (web3-based) implementation calls as soon as we get the contract address, which is what this does. If you'd like to change it to after init that's a trivial change, your call.